lib/boom_notifier/mail_notifier.ex

```diff
@@ -31,19 +31,32 @@ defmodule BoomNotifier.MailNotifier do
           {:mailer, module()} | {:from, String.t()} | {:to, String.t()} | {:subject, String.t()}
   @type options :: [option]
 
+  @impl BoomNotifier.Notifier
+  def validate!(options) do
```

Comment:
What about modeling these `validate` functions as returning a boolean value? I think it would be more natural to read and follow.

In this specific case, the code would look:

```ex
Enum.all?([:mailer, :from, :to, :subject], fn key ->
  Keyword.has_key?(options, key)
end)
```

Reply:
I like this idea 💪
However, we should come up with a way to warn the user which notifier is not valid.

Reply:
With the exception of `fetch!` we know which key is missing, if `validate` only returns a boolean, this information is not available

Reply:
With the exception that throws `fetch!` We know which key is missing, if `validate` only returns a boolean, this information is not available